### DISCUSSION TITLE
How to use try_files with s3-gateway

### DISCUSSION THREAD

**Discussion by ghost** [nginxinc/nginx-s3-gateway#157] [Q&A] [answered]
Hi,

I use the s3-gateway to serve an application from my s3 bucket. In most cases it works great. Nevertheless I have a problem with some requests:

For example my applications calls 
`/cxm/workshops/xyz`

but the webserver should serve the corresponding html file
`/cxm/workshops/xyz.html`

In the past I used try_files to solve this issue, but it doesn't seem to work with the proxy_pass argument. This is the code I used before:
```
location / {
  try_files $uri $uri/ @htmlext;
}

location ~ \.html$ {
  try_files $uri =404;
}   

location @htmlext {
  rewrite ^(.*)$ $1.html last;
}
```

Any suggestions how I can solve this problem while using the s3-gateway?

Thanks in advance.

**Comment by dekobon** [comment, marked as answer]
This is less a question about the gateway then about doing rewrite rules for a proxy. I'd try your question in the [NGINX Community Slack](https://www.nginx.com/blog/nginx-wants-you-join-us-on-the-nginxcommunity-slack/).
